crates/iota-indexer/tests/rpc-tests/read_api.rs

```diff
@@ -1158,6 +1161,13 @@ fn get_total_transaction_blocks() {
 
     runtime.block_on(async move {
         indexer_wait_for_checkpoint(store, checkpoint).await;
+        // even tough when the indexer caught up with the desired checkpoint from the
+        // fullnode, sometimes when requesting the fullnode
+        // get_checkpoint(CheckpointId::SequenceNumber(checkpoint)) it can say that it
+        // does not exist, here we make sure that also the fullnode is synced
+        // with the desired checkpoint seq number
+        fullnode_wait_for_checkpoint(cluster.rpc_client(), checkpoint).await;
```

Comment:
this is a bit weird IMO. if the indexer got a checkpoint from the node, I'm wondering why the node can't provide it via `get_checkpoint(CheckpointId::SequenceNumber(checkpoint))`?

for now, would  it be possible to maybe call `fullnode_wait_for_checkpoint` directly in `indexer_wait_for_checkpoint`? as I imagine always calling both functions manually might be a bit error prone

Reply:
@kodemartin @samuel-rufi 
I've left a comment with the solution and the investigation on the issue itself to find it easily if needed in the future: https://github.com/iotaledger/iota/issues/4171#issuecomment-2500028348